src/mtype.d

```diff
@@ -7849,30 +7847,12 @@ public:
             sc2.pop();
             return e;
         }
-        if (e.op == TOKdot)
-        {
-            DotExp de = cast(DotExp)e;
-            if (de.e1.op == TOKscope)
-            {
-                assert(0); // cannot find a case where this happens; leave
-                // assert in until we do
-                // ScopeExp se = cast(ScopeExp)de.e1;
-                // s = se.sds.search(e.loc, ident);
-                // e = de.e1;
-                // goto L1;
-            }
-        }
+
         s = sym.search(e.loc, ident);
     L1:
         if (!s)
         {
-            if (sym._scope) // it's a fwd ref, maybe we can resolve it
-            {
-                sym.semantic(null);
-                s = sym.search(e.loc, ident);
-            }
-            if (!s)
-                return noMember(sc, e, ident, flag);
+            return noMember(sc, e, ident, flag);
```

Comment:
This is not future work. It's already proven in other places where the `StructDeclaration.search` is called only once. The double `search` call was added in very old ages, and it's not necessary today. It's a time to remove the old age garbage.